contracts/modules/staking/StakingPool.sol

```diff
@@ -863,100 +853,172 @@ contract StakingPool is IStakingPool, ERC721 {
+      uint trancheId = _firstActiveTrancheId + i;
+      uint activeCoverAmount = storedAllocations[i];
+
+      if (trancheId >= firstTrancheIdToUse) {
+        trancheAllocations[trancheId - firstTrancheIdToUse] = activeCoverAmount;
+        requestedTranchesCapacityUsed += activeCoverAmount;
       }
-    }
 
-    for (uint i = 0; i < trancheCount; i++) {
-      uint activeCoverAmount = coverAmounts[i].activeCoverAmount();
-      allocatedCapacities[i] = activeCoverAmount;
-      allocatedCapacity += activeCoverAmount;
+      totalCapacityUsed += activeCoverAmount;
     }
 
-    return (allocatedCapacities, allocatedCapacity);
+    return (trancheAllocations, requestedTranchesCapacityUsed, totalCapacityUsed);
   }
 
-  function getTotalCapacitiesForActiveTranches(uint productId, uint24 globalCapacityRatio, uint16 capacityReductionRatio) public view returns (uint[] memory totalCapacities, uint totalCapacity) {
+  function getActiveTrancheCapacities(
+    uint productId,
+    uint globalCapacityRatio,
+    uint capacityReductionRatio
+  ) public view returns (
+    uint[] memory trancheCapacities,
+    uint totalCapacity
+  ) {
+
     uint firstTrancheIdToUse = block.timestamp / TRANCHE_DURATION;
 
-    (totalCapacities, totalCapacity) = getTotalCapacitiesForTranches(
+    (trancheCapacities, /* requestedTranchesCapacity */, totalCapacity) = getTrancheCapacities(
       productId,
       firstTrancheIdToUse,
       MAX_ACTIVE_TRANCHES,
       globalCapacityRatio,
       capacityReductionRatio
     );
+
+    return (trancheCapacities, totalCapacity);
   }
 
-  function getTotalCapacitiesForTranches(
+  function getTrancheCapacities(
     uint productId,
     uint firstTrancheId,
     uint trancheCount,
     uint capacityRatio,
     uint reductionRatio
-  ) internal view returns (uint[] memory totalCapacities, uint totalCapacity) {
+  ) internal view returns (
+    uint[] memory trancheCapacities,
+    uint requestedTranchesCapacity,
+    uint totalCapacity
+  ) {
 
-    uint _activeStake = Math.divCeil(activeStake, 1e12);
+    uint _activeStake = activeStake;
     uint _stakeSharesSupply = stakeSharesSupply;
+    trancheCapacities = new uint[](trancheCount);
 
     if (_stakeSharesSupply == 0) {
-      totalCapacities = new uint[](trancheCount);
-      totalCapacity = 0;
-      return (totalCapacities, totalCapacity);
+      return (trancheCapacities, 0, 0);
     }
 
-    uint weight = products[productId].targetWeight;
+    uint multiplier =
+      capacityRatio
+      * (CAPACITY_REDUCTION_DENOMINATOR - reductionRatio)
+      * products[productId].targetWeight;
+
+    uint denominator =
+      GLOBAL_CAPACITY_DENOMINATOR
+      * CAPACITY_REDUCTION_DENOMINATOR
+      * WEIGHT_DENOMINATOR;
+
+    uint lastTrancheId = (block.timestamp / TRANCHE_DURATION) + MAX_ACTIVE_TRANCHES - 1;
+
+    for (
+      uint trancheId = block.timestamp / TRANCHE_DURATION;
+      trancheId <= lastTrancheId;
+      trancheId++
+    ) {
+
+      uint trancheCapacity =
```

Comment:
with a comment or a documentation at the function level i'd want it to be more clear what values are scaled at 2 decimal levels